lib/v2/bloomberg/utils.js

```diff
@@ -0,0 +1,303 @@
+        const attachment = Object.values(story_json.imageAttachments)[0];
+        if (attachment) {
+            const image = {
+                src: attachment.baseUrl,
+                alt: attachment.alt,
+            };
+            return art(path.join(__dirname, 'templates/image_figure.art'), image);
+        }
+        return '';
+    }
+};
+
+const processVideo = async (bmmrId, summary) => {
+    const api = `https://www.bloomberg.com/multimedia/api/embed?id=${bmmrId}`;
+    const res = await got(api, { headers });
+    if (res && res.data) {
+        const video_json = res.data;
+        return {
+            stream: video_json.streams ? video_json.streams[0]?.url : '',
+            mp4: video_json.downloadURLs ? video_json.downloadURLs['600'] : '',
+            coverUrl: video_json.thumbnail.baseUrl ?? '',
+            caption: video_json.description || video_json.title || summary,
+        };
+    }
+    return {};
+};
+
+const processBody = async (body_html, story_json) => {
+    const removeSel = ['meta', 'script', '*[class$="-footnotes"]', '*[class$="for-you"]', '*[class$="-newsletter"]', '*[class$="page-ad"]', '*[class$="-recirc"]', '*[data-ad-placeholder="Advertisement"]'];
+
+    const $ = cheerio.load(body_html);
+    removeSel.forEach((sel) => $(sel).remove());
+    $('.paywall').removeAttr('class');
+
+    for (const e of $('figure')) {
+        const imageType = $(e).data('image-type');
+        const type = $(e).data('type');
+
+        let new_figure = '';
+        if (imageType === 'audio') {
+            let audio = {};
+            if (story_json.audios) {
+                const attachment = story_json.audios.find((a) => a.id.toString() === $(e).data('id').toString());
+                audio = {
+                    img: attachment.image?.url || $(e).find('img').attr('src'),
+                    src: attachment.url || $(e).find('audio source').attr('src'),
+                    caption: $(e).find('[class$="text"]').html()?.trim() ?? '',
+                    credit: $(e).find('[class$="credit"]').html()?.trim() ?? '',
+                };
+            }
+            if (story_json.episode) {
+                const episode = story_json.episode;
+                audio = {
+                    src: episode.url,
+                    img: episode.image || story_json.pageInfo.image.url,
+
+                    caption: episode.description || ($(e).find('[class$="text"]').html()?.trim() ?? ''),
+                    credit: (episode.credits.map((c) => c.name).join(', ') ?? []) || ($(e).find('[class$="credit"]').html()?.trim() ?? ''),
+                };
+            }
+            new_figure = art(path.join(__dirname, 'templates/audio_media.art'), audio);
+        } else if (imageType === 'video') {
+            if (story_json.videoAttachments) {
+                const attachment = story_json.videoAttachments[$(e).data('id')];
+                // eslint-disable-next-line no-await-in-loop
```

Comment:
You can use [for await...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of) loop